### ISSUE TITLE
Question: Does PhotoPrism support image recognition on the client side?

### ISSUE THREAD

**Issue by fakuivan** [photoprism/photoprism#1454]
I'm planning on running phototropism on a "weak" server, so it would be nice if the photo processing could be done by the clients as the images are being uploaded. This would also help with scalability, as the image processing is moved to the edge.

**Comment by lastzero** [comment]
It has a REST API, so you can upload and set labels as a developer.

**Comment by alexislefebvre** [comment]
To do this, a client would need to have TensorFlow with its data?

**Comment by fakuivan** [comment]
@alexislefebvre from what I understand tensorflow.js can be run in the browser, this could allow the web client to upload the images and the metadata, decentralizing the image processing step.

**Comment by lastzero** [comment]
Yes, but that's fragile (what if a browser restarts after an update?) and should be much slower as well unless your server is like an Intel Pentium first generation.

**Comment by fakuivan** [comment]
I don't think the real value here is in proving a faster way to tag the pictures (we also have no performance numbers), but to have something that scales better as more users join the server. Having this also prevent the server from being bogged down if a user decides to upload many pictures is a nice bonus.
Version mismatches could be avoided by embedding the front end API version along with the recognition results.

**Comment by lastzero** [comment]
Distributed systems are hard. Could make a research project.

**Comment by alexislefebvre** [comment]
> @alexislefebvre from what I understand tensorflow.js can be run in the browser, this could allow the web client to upload the images and the metadata, decentralizing the image processing step.

What is the volume of the data data that will have to be put on the client, including Tensorflow and its data used to recognize the images?
